ch_pipeline/processing/base.py

```diff
@@ -735,6 +739,106 @@ def slurm_split(line):
     return entries
 
 
+def check_duplicates(user: Optional[str] = None, cancel: bool = False):
+    """Return a list of job IDs corresponding to duplicate jobs.
+
+    Parameters
+    ----------
+    user
+        The user account to check
+    cancel
+        If True, cancel duplicate jobs. If any jobs are running, the most
+        most recently started job is kept. Otherwise, the most recently queued
+        job is kept. The most recent job is kept rather than the oldest to avoid
+        cases where a file has been modified by the newer job.
+
+    Returns
+    -------
+    duplicates
+        List of JOBIDs corresponding to duplicate pipeline jobs. If `cancel` is
+        True, these jobs will be cancelled.
+    """
+    jobs = slurm_jobs(user=user)
+
+    names = {}
+    duplicates = []
+
+    for job in jobs:
+        name = job["NAME"]
+        id = job["JOBID"]
+        status = job["ST"]
+        runtime = job["TIME_LEFT"]
+        subtime = job["SUBMIT_TIME"]
+
+        existing = names.get(name)
+
+        if existing is None:
+            # First time we've seen this job name
+            names[name] = (id, status, runtime, subtime)
+            continue
+
+        if existing[1] == "R":
+            # Both jobs are running. Keep the one that
+            # started most recently
```

Comment:
Is the idea here the most recent copy of a job is overwriting the output of the older copy, so it's the only version that can now produce a completed output?

Reply:
Yeah, that's the idea

Reply:
Makes sense.